relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -127,8 +201,11 @@ impl NotCondition {
     fn supported(&self) -> bool {
         self.inner.supported()
     }
-    fn matches<T: FieldValueProvider>(&self, value_provider: &T) -> bool {
-        !self.inner.matches(value_provider)
+    fn matches_event(&self, event: &Event, ip_addr: Option<IpAddr>) -> bool {
```

Comment:
In a follow-up, let's replace all these `ip_addr` with RequestMeta: 
- It's to be expected that we require more information about the request
- `ip_addr` doesn't specify which IP we want: the socket, the client, how to handle x-forwarded-for? This is what RequestMeta already solves for us.